### PR TITLE
feat(lock): cap-lock — shadow/bypass/tolerance field-lock policy (Spec 63 PR-2)

### PR DESCRIPTION
**Pull request by laofahai**
## What

New capability **`addons/lock/cap-lock`** — Spec 63 Phase 3 PR-2. The advanced, opt-in policy layer on top of core's Phase-1 field-lock enforcement, consuming the `field-lock-check` interceptor seam landed in #445.

It registers a `field-lock-check` interceptor (`extensions.interceptors`, `{ point: "field-lock-check", capability: "lock", handler }`) that may suppress computed `FieldLockViolation[]` only under an explicit, audited policy (Spec 63 §4.2):

| Knob | Behavior |
|------|----------|
| `shadowMode` | observe + audit-log violations, do not block (rollout observation) |
| `bypassGroups: string[]` | if `actor.groups` intersects the list → override locks (audited) |
| `toleranceMs: number` | records younger than the window are freely editable |

Evaluation is first-match-wins (shadow → bypass → tolerance); on a match it audit-logs and returns `[]` (allow).

## Security posture (this capability can weaken an enforcement boundary, so:)

- **Fail-closed default**: empty config is a pure no-op — the handler returns the violation set **unchanged** whenever no knob matches, so core re-throws with full per-field detail.
- **Never mutates** its argument (core already hands a deep clone; we still return a fresh/unmodified array — covered by a test).
- **Every suppression is audit-logged** via the injected `Logger` with `{ capability, reason, entity, actorId, actorType, tenantId, fields, violationCount }` (Spec 63 §4.2 audit trail). No logger → silent, but the policy decision is unchanged.
- **Tolerance fails closed** on a missing/unparseable `created_at` (parsed as Date / ISO / epoch). Note: a *future* `created_at` (clock skew) counts as within-window — `created_at` is a server-managed system field, so this is acceptable; flagging for reviewer awareness.
- Static `capLock` export uses default (no-op) config; real wiring uses `createCapLock({ config, logger })` (mirrors cap-permission's programmatic factory). The dev bootstrap must inject the logger for the audit trail.

## Core change

Adds an optional `interceptors` key to the capability manifest schema (`extensionManifestSchema`) so a capability can declaratively name the interceptor points it attaches to. cap-lock is the first interceptor-using capability; previously only `hooks` existed (a distinct event-reaction concept). Additive + backward-compatible; the manifest `extensions` block is declarative-only (no code exhaustively iterates its keys).

## Out of scope (separate follow-up PRs)

§5.2 cap-lock UI extensions (lock icon / tooltip / unlock button), SOFT_LOCK two-step confirmation, notification integration.

## Test plan

- `bun test addons/lock/cap-lock/__tests__/` → **29 pass / 0 fail** (interceptor behavior incl. fail-closed + no-mutation + tolerance edge cases; capability metadata/registration; config defaults/validation).
- Core `capability-metadata` schema test green with the new key.
- `bun run typecheck` + `bun run check` clean.
- `linch lint-capability addons/lock/cap-lock` → all checks pass (§9.1).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: addons/lock/cap-lock/src/factory.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  ];
+
+  return defineCapability({
+    name: "cap-lock",
+    label: "Field Lock Policy",
+    description:
+      "Advanced field-lock policy: shadow mode, bypass groups, tolerance period, and audit trail over core field-lock enforcement",
+    type: "standard",
+    category: "system",
+    version: "0.0.1",
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/lock/cap-lock/src/factory.ts]: Fixed in ccfebf2 — factory CapabilityDefinition `version` is now `1.0.0`, matching package.json and capability.json.

File: addons/lock/cap-lock/__tests__/capability.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+function makeViolations(): FieldLockViolation[] {
+  return [{ field: "amount", type: "locked", message: 'Field "amount" is locked' }];
+}
+
+describe("cap-lock capability metadata", () => {
+  it("has the expected metadata", () => {
+    expect(capLock.name).toBe("cap-lock");
+    expect(capLock.type).toBe("standard");
+    expect(capLock.category).toBe("system");
+    expect(capLock.version).toBe("0.0.1");
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/lock/cap-lock/__tests__/capability.test.ts]: Fixed in ccfebf2 — the assertion now expects `1.0.0`.

File: addons/lock/cap-lock/src/field-lock-interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    // 2. Bypass groups — the actor's groups/roles override locks.
+    //    The core `Actor.groups: string[]` field carries the actor's
+    //    group/role memberships (set by the auth/permission slots — see
+    //    cap-permission's `actor.groups` usage), which is exactly the concept
+    //    the spec example references as `context.actor.groups`. Use it directly.
+    if (
+      policy.bypassGroups.length > 0 &&
+      context.actor.groups.some((group) => policy.bypassGroups.includes(group))
```
[the remark this answers is not in the document]
Comment (laofahai) [reply addons/lock/cap-lock/src/field-lock-interceptor.ts]: Fixed in ccfebf2 — `context.actor.groups?.some(...)`. The core `Actor.groups` type is required (`string[]`), but a system actor / API key could be constructed without it at runtime; optional chaining makes undefined short-circuit the `&&`, so it falls through to no-bypass (fail-closed) rather than throwing. typecheck + biome stay clean.
